tests/v2/validation/auth/userretention/userretention_delete_user_test.go

```diff
@@ -0,0 +1,262 @@
+	defer subSession.Cleanup()
+	setupUserRetentionSettings(ur.client, "", "400h", "*/1 * * * *", "false")
+
+	logrus.Info("Getting admin user details")
+	adminID, err := users.GetUserIDByName(ur.client, "admin")
+	require.NoError(ur.T(), err)
+	adminUser, err := ur.client.Management.User.ByID(adminID)
+	require.NoError(ur.T(), err)
+	adminUser.Password = ur.client.RancherConfig.AdminPassword
+
+	logrus.Info("Updating user attributes")
+	userAttributes, err := ur.client.WranglerContext.Mgmt.UserAttribute().Get(adminUser.ID, v1.GetOptions{})
+	require.NoError(ur.T(), err)
+	userAttributes.DeleteAfter = &v1.Duration{Duration: time.Second * 10}
+	_, err = ur.client.WranglerContext.Mgmt.UserAttribute().Update(userAttributes)
+	require.NoError(ur.T(), err)
+
+	logrus.Info("Attempting initial login")
+	_, err = auth.GetUserAfterLogin(ur.client, *adminUser)
+	require.NoError(ur.T(), err)
+
+	logrus.Info("Waiting for default duration")
+	time.Sleep(defaultWaitDuration)
+
+	logrus.Info("Attempting login after wait period")
+	_, err = auth.GetUserAfterLogin(ur.client, *adminUser)
+	require.NoError(ur.T(), err)
+}
+
+func (ur *URDeleteTestSuite) TestAdminUserGetDeleted() {
+	logrus.Info("Setting up user retention settings")
+	subSession := ur.session.NewSession()
+	defer subSession.Cleanup()
+	setupUserRetentionSettings(ur.client, "", "400h", "*/1 * * * *", "false")
+
+	logrus.Info("Creating new admin user")
+	newAdminUser, err := users.CreateUserWithRole(ur.client, users.UserConfig(), "admin")
+	require.NoError(ur.T(), err)
+	userReLogin, err := auth.GetUserAfterLogin(ur.client, *newAdminUser)
+	require.NoError(ur.T(), err)
+
+	logrus.Info("Updating user attributes")
+	userAttributes, err := ur.client.WranglerContext.Mgmt.UserAttribute().Get(newAdminUser.ID, v1.GetOptions{})
+	require.NoError(ur.T(), err)
+	userAttributes.DeleteAfter = &v1.Duration{Duration: time.Second * 10}
+	_, err = ur.client.WranglerContext.Mgmt.UserAttribute().Update(userAttributes)
+	require.NoError(ur.T(), err)
+
+	logrus.Info("Verifying user status")
+	userReLogin, err = ur.client.WranglerContext.Mgmt.User().Get(userReLogin.Name, v1.GetOptions{})
+	require.Equal(ur.T(), isActive, *(userReLogin).Enabled, "User's active state does not match expected state")
+	require.NoError(ur.T(), err)
+
+	logrus.Info("Waiting for default duration")
+	pollUserStatus(ur.client, newAdminUser.ID, isInActive)
+
+	logrus.Info("Attempting login after wait period")
+	_, err = auth.GetUserAfterLogin(ur.client, *newAdminUser)
+	require.ErrorContains(ur.T(), err, "401 Unauthorized")
+
+	logrus.Info("Checking bindings after user deletion")
+	bindingsAfter, err := rbac.GetBindings(ur.client, newAdminUser.ID)
+	require.NoError(ur.T(), err)
+	assert.Empty(ur.T(), bindingsAfter["RoleBindings"], "Expected no RoleBindings after user deletion")
+	assert.Equal(ur.T(), 0, len(bindingsAfter["RoleBindings"].([]rbac1.RoleBinding)), "RoleBindings slice should be empty after user deletion")
```

Comment:
Do we need this check? It seems redundant because you are already checking if the slice is empty in the previous line